includes/cart/functions.php

```diff
@@ -122,73 +122,118 @@ function edd_add_to_cart( $download_id, $options = array() ) {
 
 	$download = get_post( $download_id );
 
-	if( 'download' != $download->post_type )
+	if ( 'download' != $download->post_type )
 		return; // Not a download product
 
-	if ( ! current_user_can( 'edit_post', $download->ID ) && ( $download->post_status == 'draft' || $download->post_status == 'pending' ) )
+	if ( ! current_user_can( 'edit_post', $download->ID ) && ( $download->post_status == 'draft' || $download->post_status == 'pending' ) ) {
 		return; // Do not allow draft/pending to be purchased if can't edit. Fixes #1056
+	}
+
+	// Support multiple price id's being added on a single edd_add_to_cart call
+	if ( isset( $options[ 'price_id' ] ) && is_array( $options[ 'price_id' ] ) ) {
+		$last_cart_count = 0;
+		$quantity = array();
+		if ( isset( $options[ 'quantity' ] ) && is_array( $options[ 'quantity' ] ) ) {
+			$quantity = $options[ 'quantity' ];
+		} else if ( isset( $options[ 'quantity' ] ) ) {
+				$quantity = array( $options[ 'quantity' ] );
+			} else {
+			$quantity = array( 1 );
+			$options[ 'quantity' ] = $quantity;
+		}
+
+		if ( is_array( $options[ 'quantity' ] ) ) {
+			// if the number of quantities is the same as the number of price options, match them up
+			if ( count( $options[ 'quantity' ] ) == count( $options[ 'price_id' ] ) ) {
+				foreach ( edd_combine_array( $options[ 'price_id' ], $options[ 'quantity' ] ) as $price => $quantity ) {
+					$local_options = $options;
+					$local_options[ 'price_id' ] = preg_replace( '/[^0-9\.-]/', '', $price );
```

Comment:
Instead of regular expressions here. I'd say it's safer to run an `is_numeric` check on this, and make sure that `absint` matches the `(int)`.

I'd act on caution when parsing to get a price ID if a string was sent in. Worse case scenario their items are not added to the cart.